src/kernels/raw/finder/preferredRemoteKernelIdProvider.ts

```diff
@@ -31,7 +32,7 @@ export class PreferredRemoteKernelIdProvider {
         const list: KernelIdListEntry[] = this.globalMemento.get<KernelIdListEntry[]>(ActiveKernelIdList, []);
         if (list) {
             // Not using a map as we're only going to store the last 40 items.
-            const fileHash = this.crypto.createHash(uri.toString(), 'string');
+            const fileHash = this.crypto.createHash(getComparisonKey(uri), 'string');
             const entry = list.find((l) => l.fileHash === fileHash);
```

Comment:
Won't this cause issues in desktop now.
To ensure this works in desktop (previous versions) & new versions, I think the change should be to compare against the new hash & the old hash (using `uri.toString` & `getComparisonKey`)

Reply:
Yeah the old hashes won't be there. But then they just have to pick a kernel again once. The problem with uri.toString() is that it's not case insensitive.

Reply:
I put this back the way it was before. It was working before, so why switch it.